views/components/ChartList.jsx

```diff
@@ -45,54 +51,117 @@ const ChartList = () => {
       console.error(error, '*****')
     }
   }
+  const handleShareChart = (chart) => setChartToShare(chart)
+  const shareWithUsers = async (chart_id, sharedWith) => {
+    try {
+      const {
+        data: { ok },
+      } = await shareChart(chart_id, sharedWith)
+
+      if (ok === 1) {
+        const { data } = await getCharts()
+        setChartList(data)
+        setChartToShare({})
+      }
+    } catch (error) {}
+  }
+  const handleChange = (event) =>
+    setChartToShare((chart) => ({ ...chart, sharedWith: event.target.value }))
+  const handleDelete = (userToDelete) => {
+    setChartToShare((chart) => ({
+      ...chart,
+      sharedWith: chart.sharedWith.filter(
+        (username) => username !== userToDelete
+      ),
+    }))
+  }
+  const closeDialog = () => {
+    setChartToShare({})
+  }
+
+  useEffect(() => {
+    getCharts().then((res) => setChartList(res.data))
+    fetchUsernames().then((data) => {
+      const usernames = data
+        .map((username) => ({
+          value: username,
+          label: username,
+        }))
+        .filter(({ value }) => value !== user.uid)
+      setUsers(usernames)
+    })
+  }, [])
```

Comment:
we should be loading this data at the view layer and passing this down as props to follow the convention we set in other pages.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/131/commits/e250b027af6997da38ebfdf1d05fb1a75107db72